src/custom/state/gnosisSafe/updater.tsx

```diff
@@ -1,29 +1,27 @@
 import { useEffect } from 'react'
-import { useAppDispatch } from 'state/hooks'
-import { setSafeInfo } from './actions'
 import { useWalletMetaData } from 'hooks/useWalletInfo'
 import { getSafeInfo } from 'api/gnosisSafe'
 import { useWeb3React } from 'web3-react-core'
+import { useAtom } from 'jotai'
+import { gnosisSafeAtom } from 'state/gnosisSafe/atoms'
 
 const GNOSIS_SAFE_WALLET_NAMES = ['Gnosis Safe Multisig', 'Gnosis Safe', 'Gnosis Safe App']
 
 export default function Updater(): null {
-  const dispatch = useAppDispatch()
   const { account, chainId, library } = useWeb3React()
   const { walletName } = useWalletMetaData()
+  const [, setGnosisSafeInfo] = useAtom(gnosisSafeAtom)
```

Comment:
There's also a shortcut for that :) 

https://jotai.org/docs/utils/use-update-atom